### ISSUE TITLE
Tensorflow & Keras version

### ISSUE THREAD

**Issue by sungkwang** [sciann/sciann#4]
Hello, I would like to know what version of Tensorflow and Keras should be used. For Tensorflow, I tried version 1.15 and 2.2, but I've got the following errors.
<1.15>
ImportError: Keras requires TensorFlow 2.2 or higher. Install TensorFlow via `pip install tensorflow`
<2.2>
ImportError: cannot import name 'is_tensor' from 'keras.backend' (C:\Program_Files\anaconda3\envs\tf2-gpu\lib\site-packages\keras\backend.py)

I am using Windows 10 and anaconda. Tensorflow works fine. And, I installed sciann using pip as recommended here. Thank you.

**Comment by imsenthur** [comment]
Tested with **Tensorflow 2.2.0 and Keras 2.4.3**

The issue seems to be with '_sciann/utils/utilities.py_', replace the following lines of code.
https://github.com/sciann/sciann/blob/e31b25de7577c409596d982725944178b891988a/sciann/utils/utilities.py#L16 to `from tensorflow import is_tensor`

https://github.com/sciann/sciann/blob/e31b25de7577c409596d982725944178b891988a/sciann/utils/utilities.py#L22-L25 to,
```
from keras.initializers import RandomUniform as default_bias_initializer
from keras.initializers import GlorotNormal as default_kernel_initializer
from keras.initializers import Constant as default_constant_initializer
from keras.regularizers import l1_l2
```
It should be up and running.

**Comment by sungkwang** [comment]
Works now. Thanks.

**Comment by sungkwang** [comment]
I faced another problem. The error looks related to keras. 

Traceback (most recent call last):
  File "SciANN-BurgersEquation.py", line 62, in <module>
    u = sn.Functional('u', [t,x], 8*[20], 'tanh')
  File "C:\Program_Files\anaconda3\envs\tf2-gpu\lib\site-packages\sciann\functionals\functional.py", line 145, in __init__
    layer.name = "conct_" + layer.name.split("_")[-1]
  File "C:\Program_Files\anaconda3\envs\tf2-gpu\lib\site-packages\tensorflow\python\keras\engine\base_layer.py", line 2554, in __setattr__
    'different name.').format(name))
AttributeError: Can't set the attribute "name", likely because it conflicts with an existing read-only @property of the object. Please choose a different name.

Using Tensorflow 2.2.0 and Keras 2.4.3.

**Comment by imsenthur** [comment]
It should be `layer._name = 'foo'`, since `layer.name` is read-only, according to 
[https://stackoverflow.com/questions/49550182/keras-rename-model-and-layers](url)

So I modified '_\sciann\functionals\functional.py_' and '_\sciann\utils\math.py_' which seemed to fix it. The 1D curve-fitting example: https://github.com/sciann/sciann/blob/master/examples/example-fitting-1d.py worked as expected, but others did not. There are few problems during eager execution [https://github.com/tensorflow/tensorflow/issues/34944](url).

Maybe it doesn't support Tensorflow 2.2.0. I'll try it out with >=1.14.

**Comment by imsenthur** [comment]
Works on Tensorflow 1.14.0. Downgrade your Keras version to 2.3.0 with `pip install --upgrade keras==2.3.0`.

**Comment by dimitri-voytan** [comment]
> Works on Tensorflow 1.14.0. Downgrade your Keras version to 2.3.0 with `pip install --upgrade keras==2.3.0`. You can also try out Tensorflow 1.15.0 with Keras 2.3.1. Kindly refer to https://docs.floydhub.com/guides/environments/.

This was a straightforward workaround for me using Google colab notebooks. Thank you.

**Comment by dimitri-voytan** [comment]
I was able to get around this issue using Tensorflow 1.15.2. I am using Google Colab notebooks. Tensorflow v2.3 is imported by default and I get the same problem as @sungkwang . These imports appear to work, but I can't verify that they aren't broken in a subtle way. 

`%tensorflow_version 1.x
import tensorflow as tf
!pip install sciann`

**Comment by sciann** [comment]
please use TF <= 2.2.0 and Keras==2.3.1. A new release is coming soon that support the latest release of tensorflow.

**Comment by sciann** [comment]
Should be resolved now - get the latest v0.5.0.0 version.
